crates/libtiny_tui/src/tests/mod.rs

```diff
@@ -443,6 +443,50 @@ fn test_alignment_long_string() {
 
     expect_screen(screen, &tui.get_front_buffer(), 40, 5, Location::caller());
 }
+
+#[test]
+fn test_resize_recalc_scroll() {
+    let mut tui = TUI::new_test(15, 5);
+    let serv = "irc.server_1.org";
+    let chan = ChanNameRef::new("#chan");
+    tui.new_server_tab(serv, None);
+    tui.set_nick(serv, "osa1");
+    tui.new_chan_tab(serv, chan);
+    tui.next_tab();
+    tui.next_tab();
+
+    let target = MsgTarget::Chan { serv, chan };
+    let ts = time::at_utc(time::Timespec::new(0, 0));
+    tui.add_privmsg(
+        "osa1",
+        "1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111 1111",
+        ts,
+        &target,
+        false,
+        false,
+    );
+
+    tui.draw();
```

Comment:
Would be good to see the screen before resizing, so that I can compare before and after. Currently I see the after, but I don't know what it was before, so I can't tell immediately if the test is right or wrong, and I have to study it and debug it.

Reply:
No problem, fixing this and adding more tests